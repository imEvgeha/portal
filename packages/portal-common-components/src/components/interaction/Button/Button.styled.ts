
import styled, { css, keyframes } from 'styled-components';
import { media } from '../../../styles/utils';

import { ButtonProps } from './types';

export const buttonLoaderAnimation = keyframes`
  0%, 80%, 100% {
    transform: scale(0);
  } 40% {
    transform: scale(1.0);
  }
`;

export const StyledButtonLoader = styled.div`
  width: 70px;
  display: flex;
  align-items: center;
  justify-content: center;

  & > div {
    width: 6px;
    height: 6px;
    margin-right: 3px;
    background-color: #FFFFFF;
    border-radius: 100%;
    animation: ${buttonLoaderAnimation} 1.4s infinite ease-in-out both;

    &:nth-child(1) {
      animation-delay: -0.32s;
    }

    &:nth-child(2) {
      animation-delay: -0.16s;
    }
  }
`;

export const ButtonIcon = styled.div`
  width: 16px;
  height: 16px;
  margin: 0 8px 0 0;
  display: flex;
  align-items: center;
`;

export const StyledButton = styled.button<ButtonProps>`
  color: #FFFFFF;
  background-color: #007BFF;
  font-family: sans-serif;
  font-size: 16px;
  font-weight: 500;
  width: 200px;
  height: 48px;
  padding: 12px;

  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  line-height: normal;
  text-align: center;
  text-decoration: none;
  cursor: pointer;
  outline: none;
  border-radius: 4px;
  border: 0;

  svg {
    fill: #FFFFFF;
    width: 16px;
    height: 16px;
  }

  &:hover {
    color: #FFFFFF;
    background-color: #69A9FF;
  }

  &:focus {
    background-color: #0050CB;
  }

  ${media.desktop`
    transition: background-color .2s, border-color .2s;

    + button {
      margin: 0 0 0 8px;
    }
  `}

  ${({ size }) => size === 'auto' && css`
    width: auto;
  `}

  ${({ isLoading }) => isLoading && css`
    cursor: not-allowed;

    &:hover,
    &:focus {
      background-color: #007BFF;
    }
  `}

  ${({ disabled }) => disabled && css`
    color: #FFFFFF};
    background-color: #E6E6E6;
    cursor: not-allowed;

    &:focus {
      background-color: #E6E6E6;
    }

    ${media.desktop`
      transition: none;

      &:hover {
        background-color: #E6E6E6;
      }
    `}
  `};

  ${({ iconPosition }) => iconPosition === 'right' && css`
    flex-direction: row-reverse;

    ${ButtonIcon} {
      margin: 0 0 0 8px;
    }
  `}

  ${({ iconOnly }) => iconOnly && css`
    width: 48px;

    ${ButtonIcon} {
      margin: 0;
    }
  `}
`;
