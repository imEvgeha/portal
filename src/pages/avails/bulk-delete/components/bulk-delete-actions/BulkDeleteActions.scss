@import '../../../../../styles/atlaskit/mixins';

.nexus-c-bulk-delete-actions__wrapper {
    @include margin(5, 0, 0, 0);
}

.nexus-c-bulk-delete-actions__btn-wrapper {
    width: 100%;
    display: flex;
}

.nexus-c-bulk-delete-actions__cancel-btn {
    @include margin(0, 2, 0, auto);
}
