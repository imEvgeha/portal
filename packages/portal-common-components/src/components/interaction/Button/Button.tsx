import * as React from 'react';
import { Link } from 'react-router-dom';

import { ButtonProps } from './Button.types';
import { StyledButton, StyledButtonLoader, ButtonIcon } from './Button.styled';

const ButtonLoader = () => (
  <StyledButtonLoader>
    <div />
    <div />
    <div />
  </StyledButtonLoader>
);

const ButtonContent: React.FC<ButtonProps> = ({ children, icon, isLoading }) => (
  <>
    {isLoading ? (
      <ButtonLoader />
    ) : (
      <>
        {icon && (
          <ButtonIcon>
            {icon}
          </ButtonIcon>
        )}
        <span>{children}</span>
      </>
    )}
  </>
);

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(({
  children, href, icon, iconPosition = 'left', isLoading, onClick, to, urlTarget, disabled,
  ...otherProps
}, ref) => {
  const styledButtonProps = {
    disabled,
    iconOnly: !children && icon,
    iconPosition,
    isLoading,
    ref,
  };

  const buttonContentProps = {
    children,
    icon,
    isLoading,
  };

  if (href) {
    return (
      <StyledButton
        as="a"
        href={href}
        target={`_${urlTarget || 'self'}`}
        {...styledButtonProps}
        {...otherProps}
      >
        <ButtonContent {...buttonContentProps} />
      </StyledButton>
    );
  }

  if (to) {
    return (
      <StyledButton
        as={Link}
        to={to}
        {...styledButtonProps}
        {...otherProps}
      >
        <ButtonContent {...buttonContentProps} />
      </StyledButton>
    );
  }

  return (
    <StyledButton
      onClick={onClick}
      {...styledButtonProps}
      {...otherProps}
    >
      <ButtonContent {...buttonContentProps} />
    </StyledButton>
  );
});
