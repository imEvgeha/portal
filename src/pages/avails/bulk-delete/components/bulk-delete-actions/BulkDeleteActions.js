import React from 'react';
import PropTypes from 'prop-types';
import Button, {LoadingButton} from '@atlaskit/button';
import {BULK_DELETE_BTN_CANCEL} from '../../constants';
import './BulkDeleteActions.scss';

const BulkDeleteActions = ({onClose, onSubmit, rightsDeletionCount, isLoading, isDisabled}) => {
    return (
        <div className="nexus-c-bulk-delete-actions__wrapper">
            <div className="nexus-c-bulk-delete-actions__btn-wrapper">
                <LoadingButton
                    appearance="danger"
                    onClick={onSubmit}
                    className="nexus-c-bulk-delete-actions__delete-btn"
                    isLoading={isLoading}
                    isDisabled={isDisabled}
                >
                    {`Delete ${rightsDeletionCount} Rights`}
                </LoadingButton>
                <Button
                    appearance="subtle"
                    onClick={onClose}
                    className="nexus-c-bulk-delete-actions__cancel-btn"
                    isDisabled={false}
                >
                    {BULK_DELETE_BTN_CANCEL}
                </Button>
            </div>
        </div>
    );
};

BulkDeleteActions.propTypes = {
    onClose: PropTypes.func,
    onSubmit: PropTypes.func,
    rightsDeletionCount: PropTypes.number,
    isLoading: PropTypes.bool,
    isDisabled: PropTypes.bool,
};

BulkDeleteActions.defaultProps = {
    onClose: () => null,
    onSubmit: () => null,
    rightsDeletionCount: 0,
    isLoading: false,
    isDisabled: false,
};

export default BulkDeleteActions;
