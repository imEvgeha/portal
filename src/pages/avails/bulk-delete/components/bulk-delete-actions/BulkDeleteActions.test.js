import React from 'react';
import Button from '@atlaskit/button';
import {shallow} from 'enzyme';
import BulkDeleteActions from './BulkDeleteActions';

describe('BulkDeleteActions', () => {
    let wrapper = null;

    beforeAll(() => {
        wrapper = shallow(<BulkDeleteActions />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders Cancel and Delete buttons', () => {
        expect(wrapper.find(Button)).toHaveLength(2);
    });
});
