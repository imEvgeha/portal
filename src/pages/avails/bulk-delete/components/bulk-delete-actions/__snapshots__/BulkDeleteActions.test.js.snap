// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkDeleteActions should match snapshot 1`] = `
<div
  className="nexus-c-bulk-delete-actions__wrapper"
>
  <div
    className="nexus-c-bulk-delete-actions__btn-wrapper"
  >
    <WithAnalyticsContext(WithAnalyticsEvents(Button))
      appearance="danger"
      className="nexus-c-bulk-delete-actions__delete-btn"
      isDisabled={false}
      onClick={[MockFunction]}
    >
      Delete 0 Rights
    </WithAnalyticsContext(WithAnalyticsEvents(Button))>
    <WithAnalyticsContext(WithAnalyticsEvents(Button))
      appearance="subtle"
      className="nexus-c-bulk-delete-actions__cancel-btn"
      isDisabled={false}
      onClick={[MockFunction]}
    >
      Cancel
    </WithAnalyticsContext(WithAnalyticsEvents(Button))>
  </div>
</div>
`;
